huvr_client/client.py

```diff
@@ -0,0 +1,154 @@
+from .api.assessments import AssessmentsApiModule
+from .api.asset_types import AssetTypesApiModule
+from .api.assets import AssetsApiModule
+from .api.auth import AuthApiModule
+from .api.checklist_result_lines import ChecklistResultLinesApiModule
+from .api.checklist_tasks import ChecklistTasksApiModule
+from .api.checklist_templates_fillable import ChecklistTemplatesFillableApiModule
+from .api.checklist_templates import ChecklistTemplatesApiModule
+from .api.checklists import ChecklistsApiModule
+from .api.cmls import CmlsApiModule
+from .api.companies import CompaniesApiModule
+from .api.crews import CrewsApiModule
+from .api.defect_profiles import DefectProfilesApiModule
+from .api.defects import DefectsApiModule
+from .api.heat_data import HeatDataApiModule
+from .api.inspection_media_overlays import InspectionMediaOverlaysApiModule
+from .api.inspection_media import InspectionMediaApiModule
+from .api.libraries import LibrariesApiModule
+from .api.library_media import LibraryMediaApiModule
+from .api.location_layer_profiles import LocationLayerProfilesApiModule
+from .api.measurements import MeasurementsApiModule
+from .api.observations_summary import ObservationsSummaryApiModule
+from .api.pages import PagesApiModule
+from .api.project_types import ProjectTypesApiModule
+from .api.projects import ProjectsApiModule
+from .api.recommended_work_plans import RecommendedWorkPlansApiModule
+from .api.reservations import ReservationsApiModule
+from .api.schedules import SchedulesApiModule
+from .api.users import UsersApiModule
+
+
+class HuvrClient:
+    """
+    REST API for HUVR.
+
+    See full REST API documentation at https://docs.huvrdata.com
+    """
+
+    def __init__(
+        self,
+        base_url: str,
+        verbose: bool = False,
+    ):
+        """
+        Initialize the client.
+
+        :param base_url: The base URL for the HUVR API. ex: https://demo.huvrdata.com
+        :param verbose: print debug to stdout.
+        """
+        self.base_url = base_url
+        # remove trailing slash if present
+        if self.base_url.endswith("/"):
+            self.base_url = self.base_url[:-1]
+
+        self.verbose = verbose
+        self.session = requests.Session()
+
+        # modules
+        #
+        self.assessment_types = AssessmentTypesApiModule(self)
+        self.assessments = AssessmentsApiModule(self)
+        self.asset_types = AssetTypesApiModule(self)
+        self.assets = AssetsApiModule(self)
+        self.auth = AuthApiModule(self)
+        self.checklist_result_lines = ChecklistResultLinesApiModule(self)
+        self.checklist_tasks = ChecklistTasksApiModule(self)
+        self.checklist_templates_fillable = ChecklistTemplatesFillableApiModule(self)
+        self.checklist_templates = ChecklistTemplatesApiModule(self)
+        self.checklists = ChecklistsApiModule(self)
+        self.cmls = CmlsApiModule(self)
+        self.companies = CompaniesApiModule(self)
+        self.crews = CrewsApiModule(self)
```

Comment:
this is all manual for now - prioritizing simplicity for user to understand how client works, and to be sure hints are present in IDE (not doing magic importing / dynamic setattr)

![Screenshot 2023-09-20 at 4 34 05 PM](https://github.com/huvrdata/huvr-client/assets/11606807/5c101ba3-82e9-41d5-9c29-c767a4fb6290)